core-web/libs/dotcms-js/src/lib/core/site.service.ts

```diff
@@ -70,10 +70,11 @@ export class SiteService {
         if (siteIdentifier === this.selectedSite.identifier) {
             name === 'ARCHIVE_SITE'
                 ? this.switchToDefaultSite()
-                      .pipe(take(1))
-                      .subscribe((currentSite: Site) => {
-                          this.switchSite(currentSite);
-                      })
+                      .pipe(
+                          take(1),
+                          switchMap((site) => this.switchSite(site))
```

Comment:
why this change?

Reply:
That function now returns an observable. That's why I use it inside a `switchMap`